src/mock_stream.rs

```diff
@@ -1,6 +1,7 @@
 use std::io::{Error, ErrorKind, Read, Result, Write};
 use std::cmp::min;
 
+#[derive(Copy, Clone, Debug, Eq, Ord, PartialEq, PartialOrd, Hash)]
```

Comment:
`MockStream` is not exposed outside the crate, and there's no reason for it to derive these traits.

Reply:
Also, `MockStream` cannot implement `Copy`, since it contains `Vec`s.

Reply:
Fixed the wrong `Copy` derivation. I tested locally but apparently something went wrong, sorry!

Reply:
I get your point but, these traits should be implemented eagerly, the point of the recommendation is to avoid needless discussion and abitrary decisions on what traits are and aren't ‘useful’. If want to debug tests I could well have a use for many of these traits.

Reply:
I read the recommendation as being about the crate's API (and hence, about the types that are public). The reason I am hesitant about eagerly adding these derives to internal types is that it unnecessarily limits what we can put inside them for very little gain. For externally exposed types, I totally see the advantage of eagerly implementing these traits, since the overhead for users for adding `impl`s is very high. However, for internal types, it's trivial to add derives on-demand.